ovs/lib/healthcheck.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/python
+
+# Copyright 2014 iNuron NV
+#
+# Licensed under the Open vStorage Modified Apache License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.openvstorage.org/license
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Module for HealthCheckController
+"""
+
+from ovs.celery_run import celery
+from ovs.extensions.healthcheck.openvstorage.openvstoragecluster_health_check import OpenvStorageHealthCheck
+from ovs.extensions.healthcheck.arakoon.arakooncluster_health_check import ArakoonHealthCheck
+from ovs.extensions.healthcheck.alba.alba_health_check import AlbaHealthCheck
+from ovs.log.healthcheck_logHandler import HCLogHandler
+
+module = "healthcheck"
+platform = 0
+unattended = False
+silent_mode = False
+LOGGER = HCLogHandler(unattended, silent_mode)
+
+class HealthCheckController:
+
+    @staticmethod
+    @celery.task(name='ovs.healthcheck.check_unattended')
+    def check_unattended():
+        """
+        Executes the healthcheck in UNATTENDED mode
+
+        @return: results of the healthcheck
+
+        @rtype: dict
+
+        @raises Exception (When platform is not supported)
+        """
+
+        # initialize variables as global
+        global LOGGER
```

Comment:
What's the benefit of this? Do these have to be available on module level, you might just add them on class level and then use ```HealthCheckController.<whatever>```. Or just pass in variables. If this function doesn't make much sense if it's just calling another function. Why not make a ```execute_check``` task with a few parameters like ```unattended``` and ```silent_mode```?